velox/functions/prestosql/window/Rank.cpp

```diff
@@ -41,17 +51,49 @@ class RankFunction : public exec::WindowFunction {
       vector_size_t resultOffset,
       const VectorPtr& result) override {
     int numRows = peerGroupStarts->size() / sizeof(vector_size_t);
-    auto* rawValues = result->asFlatVector<int64_t>()->mutableRawValues();
     auto* rawPeerStarts = peerGroupStarts->as<vector_size_t>();
+
+    void* rawValues = nullptr;
+    switch (type) {
+      case RankType::kRank:
+      case RankType::kDenseRank: {
+        rawValues = result->asFlatVector<int64_t>()->mutableRawValues();
+        break;
+      }
+      case RankType::kPercentRank: {
+        rawValues = result->asFlatVector<double>()->mutableRawValues();
+        break;
+      }
+    }
+
     for (int i = 0; i < numRows; i++) {
       auto start = rawPeerStarts[i];
       if (start != currentPeerGroupStart_) {
         currentPeerGroupStart_ = start;
-        rank_ += previousPeerCount_;
+        if constexpr (type == RankType::kDenseRank) {
+          rank_ += (previousPeerCount_ != 0);
```

Comment:
I find it confusing to use add a boolean to an integer. Would you use an if statement?

```
if (previousPeerCount_ > 0) {
   ++rank_;
}
```

Reply:
Sure, thanks!